Symfony/CS/Tokenizer/Tokens.php

```diff
@@ -751,6 +751,106 @@ public function getPrevMeaningfulToken($index)
     }
 
     /**
+     * Find a sequence of meaningful tokens and returns the array of their locations.
+     *
+     * @param array      $sequence      an array of tokens (same format used by getNextTokenOfKind)
+     * @param int        $start         start index, defaulting to the start of the file.
+     * @param int        $end           end index, defaulting to the end of the file.
+     * @param bool|array $caseSensitive global case sensitiveness or an array of booleans, whose keys should match
+     *                                  the ones used in $others. If any is missing, the default case-sensitive
+     *                                  comparison is used.
+     *
+     * @return array|null an array containing the tokens matching the sequence elements, indexed by their position
+     */
+    public function findSequence(array $sequence, $start = 0, $end = null, $caseSensitive = true)
+    {
+        // $end defaults to the end of the file
+        if (null === $end) {
+            $end = count($this) - 1;
+        }
+
+        if (!count($sequence)) {
+            throw new \InvalidArgumentException('Invalid sequence');
+        }
+
+        // make sure the sequence content is "meaningful"
+        foreach ($sequence as $key => $token) {
+            // if not a Token instance already, we convert it to verify the meaningfulness
+            if (!$token instanceof Token) {
+                if (is_array($token) && !isset($token[1])) {
+                    // fake some content as it is required by the Token constructor,
+                    // although optional for search purposes
+                    $token[1] = '';
+                }
+                $token = new Token($token);
+            }
+            if ($token->isWhitespace() || $token->isComment() || $token->isEmpty()) {
+                throw new \InvalidArgumentException('Non-meaningful token at position: '.$key);
+            }
+        }
+
+        // a closure to get case-sensitiveness for any arbitrary key
+        if (is_array($caseSensitive)) {
```

Comment:
In out 1.5 branch we handle this differently.
Please use one version and try do not repeat the code.